### PR TITLE
Story1reviewed

### PR DESCRIPTION
**Pull request by DanAbsolon**
trainer review

### CODE REVIEW

File: src/functions.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        && $book['authorFirstName']
+        && $book['keyQuote']
+        && $book['languageWrittenIn']
+        && $book['link']
+        && $book['extra']) {
+            if (substr($book['authorSecondName'], -1) == "s") {
+                $possessiveMarker = "'";
+            } else {
+                $possessiveMarker = "'s";
+            }
```
Comment (norayna) [inline src/functions.php:34]: this needs to be a separate function, which needs unit tests

File: src/functions.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        && $book['bookTitle']
+        && $book['textType']
+        && $book['genre']
+        && $book['yearPublished']
+        && $book['authorInspiration']
+        && $book['authorFirstName']
+        && $book['keyQuote']
+        && $book['languageWrittenIn']
+        && $book['link']
+        && $book['extra']) {
```
Comment (norayna) [inline src/functions.php]: this validation plus the exception logic should be in a separate function called something like `validateBook`, which either returns true or throws an exception. then that function needs to be unit tested as well

File: index.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+?>
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>Bookshelf</title>
+	<link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css">
+	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+	<script src="https://kit.fontawesome.com/5139af5f3f.js" crossorigin="anonymous"></script>
```
Comment (norayna) [inline index.php:19]: these 3 lines are 1 tab too deep

Comment (DanAbsolon) [reply index.php:19]: it think it's just on the github display, it's fine here

**Review by norayna — APPROVED** [review]
APPROVED

**Pull request merged**